### PR TITLE
fix(core): add watch decorator for value prop

### PR DESCRIPTION
**Pull request by luiz-ortega**
## Infos

[Task](https://juntossomosmais.monday.com/boards/1756815464/pulses/5277352428)

#### What is being delivered?

Added `Watch` decorator and `watchValueHandler` method for right reactivity behavior with onAtomChange prop. 

#### What impacts?

The prop onAtomChange runs on every value update while typing on react applications. 

#### Reversal plan

[Describe which plan we should follow if this delivery has to be reversed.](https://login.microsoftonline.com/49158df8-04df-4db2-a1ae-ee3949ca3984/oauth2/authorize?client%5Fid=00000003%2D0000%2D0ff1%2Dce00%2D000000000000&response%5Fmode=form%5Fpost&response%5Ftype=code%20id%5Ftoken&resource=00000003%2D0000%2D0ff1%2Dce00%2D000000000000&scope=openid&nonce=AC2D47141A613BA5C1B6092D4707302171E1DA38D8662F3B%2DBDF23000C02955542D2DAF0DFBC9EC27F263CFC1C00238D3ABB4772C8CB9B8CD&redirect%5Furi=https%3A%2F%2Fvotorantimindustrial%2Esharepoint%2Ecom%2F%5Fforms%2Fdefault%2Easpx&state=OD0w&claims=%7B%22id%5Ftoken%22%3A%7B%22xms%5Fcc%22%3A%7B%22values%22%3A%5B%22CP1%22%5D%7D%7D%7D&wsucxt=1&cobrandid=11bd8083%2D87e0%2D41b5%2Dbb78%2D0bc43c8a8e8a&client%2Drequest%2Did=73f0f2a0%2D803b%2D4000%2D7fb2%2D81ec3f5bef20)

### CODE REVIEW

File: packages/core/src/components/input/input.tsx [hunk 1]
```diff
@@ -120,12 +121,19 @@ export class AtomInput {
...
     const value = event.target.value
+
     this.value = value
     this.atomChange.emit(String(value))
   }
 
+  @Watch('value')
+  watchValueHandler(newValue: string) {
+    this.atomChange.emit(String(newValue))
+  }
```
Comment (felipefialho) [inline packages/core/src/components/input/input.tsx:132]: 👏

**Review by felipefialho — DISMISSED** [review]

**Comment by felipefialho** [comment]
@luiz-ortega Maybe could u add this inside `atom-select` and `atom-textarea` too?

File: packages/core/src/components/input/input.tsx [hunk 2]
```diff
@@ -144,7 +152,7 @@ export class AtomInput {
     return (
       <Host>
         <ion-input
-          ref={(el) => (this.inputEl = el as HTMLIonInputElement)}
+          ref={(el) => (this.inputEl = el)}
```
Comment (luiz-ortega) [inline packages/core/src/components/input/input.tsx:155]: Type assertion removed for sonar-lint resolution
image

**Comment by luiz-ortega** [comment]
> @luiz-ortega Maybe could u add this inside `atom-select` and `atom-textarea` too?

@felipefialho in a conversation with @igorwessel we tested this issue with the select component and the bug didn't happen :+1: 
Done on textArea https://github.com/juntossomosmais/atomium/pull/317/commits/875f3a4bc0ac57a28a8b4efe073cf3a3e354f27e

**Review by igorwessel — APPROVED** [review]

**Pull request merged**